tests/test_functions.py

```diff
@@ -0,0 +1,197 @@
+import sys
+import os
+import unittest
+import tempfile
+
+from csvjoin import functions
+from io import StringIO
+
+
+class Capturing(list):
+    """
+    Captures the output from the print() function.
+
+    Only for tests.
+    """
+    def __enter__(self):
+        self._stdout = sys.stdout
+        sys.stdout = self._stringio = StringIO()
+        return self
+
+    def __exit__(self, *args):
+        self.extend(self._stringio.getvalue().splitlines())
+        del self._stringio  # free up some memory
+        sys.stdout = self._stdout
+
+
+class MyTestCase(unittest.TestCase):
+
+    def test_parse_arguments_number_of_arguments(self):
+        # Too few arguments
+        sys.argv = ['join.py']
+        with self.assertRaises(TypeError):
+            functions.parse_arguments()
+
+        sys.argv = ['join.py', '../data/Employee.csv', '../data/MFG10YearTerminationData.csv']
+        with self.assertRaises(TypeError):
+            functions.parse_arguments()
+
+        # Too many arguments
+        sys.argv = ['join.py', '../data/Employee.csv', '../data/MFG10YearTerminationData.csv', 'Age', 'left', 'left']
+        with self.assertRaises(TypeError):
+            functions.parse_arguments()
+
+        sys.argv = ['join.py', '../data/Employee.csv', '../data/MFG10YearTerminationData.csv', 'Age', 'left', 'left',
+                    '../data/turnover.csv', 'Age', 'left', 'left']
+        with self.assertRaises(TypeError):
+            functions.parse_arguments()
```

Comment:
This is not a unit test, but rather a suit of unit tests. This should be separated.